install/kubernetes/cilium/files/spire/init.bash

```diff
@@ -0,0 +1,48 @@
+# shellcheck disable=SC2086
+# shellcheck disable=SC2139
+set -e
+
+echo "Waiting for spire process to start"
+while ! pgrep spire-server > /dev/null; do sleep 5; done
+
+SPIRE_SERVER_ROOT_PATH="/proc/$(pgrep spire-server)/root"
+
+alias spire_server="${SPIRE_SERVER_ROOT_PATH}/opt/spire/bin/spire-server"
+SOCKET_PATH="${SPIRE_SERVER_ROOT_PATH}/tmp/spire-server/private/api.sock"
+SOCKET_FLAG="-socketPath ${SOCKET_PATH}"
+
+echo "Checking spire-server status"
+while ! spire_server entry show ${SOCKET_FLAG} &> /dev/null; do
+  echo "Waiting for spire-server to start..."
+  sleep 5
+done
+
+echo "Spire Server is up, initializing cilium spire entries..."
+
+AGENT_SPIFFE_ID="spiffe://{{ .Values.auth.mTLS.spire.trustDomain }}/ns/{{ .Values.auth.mTLS.spire.install.namespace }}/sa/spire-agent"
```

Comment:
Update: I see that these are actually in cilium-config. In which case, I'd recommend pulling them from there, rather than via values; since it prevents divergence between configmap consumers and this script when changing.

Reply:
cilium-config map is in another namespace, pulling these values during runtime requires additional k8s client and related RBAC, so I think it will be more complicated.